src/VecSim/algorithms/brute_force/brute_force.h

```diff
@@ -73,3 +76,400 @@ class BruteForceIndex : public VecSimIndexAbstract {
     friend class BM_VecSimBasics_DeleteVectorBF_Benchmark;
 #endif
 };
+
+/******************************* Implementation **********************************/
+
+#include "VecSim/spaces/spaces.h"
+#include "VecSim/query_result_struct.h"
+#include "VecSim/algorithms/brute_force/bf_batch_iterator.h"
+
+#include <cstring>
+#include <cmath>
+
+using namespace std;
```

Comment:
move to the head of the file
also avoid `using namespace std` but have `using std::classname`